operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -246,4 +251,75 @@ public void testHostnameStrictDisabled() {
         }
     }
 
+    // Reference curl command:
+    // curl --insecure --data "grant_type=password&client_id=admin-cli&username=admin&password=adminPassword" https://localhost:8443/realms/master/protocol/openid-connect/token
+    @Test
+    public void testInitialAdminUser() {
+        try {
+            // Recreating the database to keep this test isolated
+            deleteDB();
+            deployDB();
+            var kc = getDefaultKeycloakDeployment();
+            deployKeycloak(k8sclient, kc, true);
+
+            var decoder = Base64.getDecoder();
+            var service = new KeycloakService(k8sclient, kc);
+            var kcAdminSecret = new KeycloakAdminSecret(k8sclient, kc);
+
+            AtomicReference<String> adminUsername = new AtomicReference<>();
+            AtomicReference<String> adminPassword = new AtomicReference<>();
+            Awaitility.await()
+                    .ignoreExceptions()
+                    .untilAsserted(() -> {
+                        Log.info("Checking secret, ns: " + namespace + ", name: " + kcAdminSecret.getName());
+                        var adminSecret = k8sclient
+                                .secrets()
+                                .inNamespace(namespace)
+                                .withName(kcAdminSecret.getName())
+                                .get();
+
+                        adminUsername.set(new String(decoder.decode(adminSecret.getData().get("username").getBytes(StandardCharsets.UTF_8))));
+                        adminPassword.set(new String(decoder.decode(adminSecret.getData().get("password").getBytes(StandardCharsets.UTF_8))));
+
+                        String url = "https://" + service.getName() + "." + namespace + ":" + Constants.KEYCLOAK_HTTPS_PORT + "/realms/master/protocol/openid-connect/token";
+                        Log.info("Checking url: " + url);
+
+                        var curlOutput = K8sUtils.inClusterCurl(k8sclient, namespace, "--insecure", "-s", "--data", "grant_type=password&client_id=admin-cli&username=" + adminUsername.get() + "&password=" + adminPassword.get(), url);
+                        Log.info("Curl Output: " + curlOutput);
+
+                        assertTrue(curlOutput.contains("\"access_token\""));
+                        assertTrue(curlOutput.contains("\"token_type\":\"Bearer\""));
+                    });
+
+            // Redeploy the same Keycloak without redeploying the Database
+            k8sclient.resource(kc).delete();
```

Comment:
Please add a check that the deployment was really deleted, mainly the admin Secret. If the cluster was slow, the operator might just re-use the old objects instead of creating new, I think.

Reply:
This `assertNotEquals` was intended to check for this condition:
https://github.com/keycloak/keycloak/pull/10661/files#diff-6c9bb3849a00b8bf0ad94a582971f6335e9cefadd546187ac395cace155b2ecaR317

do you feel is enough?

Reply:
Right, that is sufficient.

Reply:
Hopefully, it won't cause any test stability problems.